src/solvers/FR/Basis.jl

```diff
@@ -333,7 +341,8 @@ function fr_operators(N, sol_pts, cor_fun)
         @assert false
     end
 
-    bl, br = zeros(nd), zeros(nd)
+    T = eltype(xg)
+    bl, br = zeros(T, nd), zeros(T, nd)
     for i in 1:nd
         bl[i] = 2.0 * dgl(N, 2.0 * xg[i] - 1.0)
```

Comment:
Is this, despite '2.0', still going to give general real types, or does it force it to be Float64? If it is not general, fix it here and everywhere. But, if it is already general, do not change anything.